front-end/src/pages/SettingsPage.tsx

```diff
@@ -0,0 +1,84 @@
+import axios from 'axios';
+import React, { useState } from 'react';
+import {
+    Card,
+    CardBody,
+    Button,
+    Form,
+    FormGroup,
+    Input,
+    Label,
+} from 'reactstrap';
+
+const SettingsPage = (loggedInUser: any) => {
+    // https://reactstrap.github.io/components/form/#app
+    // https://medium.com/better-programming/easily-create-a-form-with-react-hooks-1cab17e2be0d
+    // setting state: https://stackoverflow.com/questions/45850550/accessing-data-from-axios-get-request
+
+    const initialInputState = { displayName: "", githubUrl: "" };
+    const authorUrl = process.env.REACT_APP_API_URL + "/api/author/" + loggedInUser.loggedInUser.authorId + "/";
+    const [eachEntry, setEachEntry] = useState(initialInputState);
+    const { displayName, githubUrl } = eachEntry;
+    const [unchangedData, setUnchangedData] = useState(initialInputState);
+
+    function getAuthorData() {
+        axios.get(authorUrl).then(res => {
+            setUnchangedData(res.data);
+            console.log(unchangedData)
+        }).catch(err => {
+            console.log("GET ERROR");
+        })};
+
+    const handleInputChange = (e: any) => {
+        setEachEntry({ ...eachEntry, [e.target.name]: e.target.value });
+    };
+
+    const handleSubmit = (e: any) => {
+        e.preventDefault();
+        getAuthorData();
+
+        axios.post(authorUrl, {
```

Comment:
I think we should have some sort of UI to indicate if any error has occurred, or if the update was successful. This should make our UI functional without looking at any console logs

We could have some sort of `postError` state, and show a relevant message (it can even be a <p> tag for now) if sending an update failed

Reply:
done:

On success:
<img width="1230" alt="Screen Shot 2021-02-24 at 2 04 08 AM" src="https://user-images.githubusercontent.com/32996140/108976285-9b486a00-7644-11eb-9fa5-4ae6dc55048b.png">

On empty input:
<img width="1254" alt="Screen Shot 2021-02-24 at 2 04 35 AM" src="https://user-images.githubusercontent.com/32996140/108976343-a7342c00-7644-11eb-803f-faa3c9ab8470.png">

On error (i.e. not logged in, but trying to access settings page, or if an axios request fails that sort of thing):
<img width="1260" alt="Screen Shot 2021-02-24 at 2 03 21 AM" src="https://user-images.githubusercontent.com/32996140/108976443-c03cdd00-7644-11eb-8c75-5847fc06225f.png">

Reply:
Great!